src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -89,4 +113,28 @@ public static function handleUnserializeCallback(string $class)
     {
         throw new MessageDecodingFailedException(sprintf('Message class "%s" not found during decoding.', $class));
     }
+
+    /**
+     * @internal
+     */
+    public function handleUnserializeCallbackWithClassCreation($class)
+    {
+        self::$classNotFoundDetected = true;
+
+        $parts = explode('\\', $class);
+        $class = array_pop($parts);
+        $namespace = implode('\\', $parts);
+        $code = <<<EOPHP
+            class $class
+            {
+                public string \$__WARNING__ = '⚠⚠ WARNING This class could not be unserialized. A mock has been created on the fly. ⚠⚠';
+            }
+        EOPHP;
+
+        if ($namespace) {
+            eval("namespace $namespace { $code };");
```

Comment:
I'd prefer relying on php's existing `__PHP_Incomplete_Class`.
That'd mean using unserialize instead of eval. (after doing `ini_set('unserialize_callback_func', null)` in a try/finally

Reply:
Done ✅